code/target/src/phone/java/com/adobe/marketing/mobile/Target.java

```
@@ -982,27 +982,36 @@ private static void registerResponseContentEventListener() {
-                        } else if (callback != null) {
-                            callback.call(
-                                    DataReader.optString(
-                                            eventData,
-                                            EventDataKeys.TARGET_CONTENT,
-                                            request.getDefaultContent()));
+                        try {
+                            if (contentWithDataCallback != null) {
+                                final Map<String, Object> mboxPayloadMap =
+                                        createMboxPayloadMap(
+                                                DataReader.optTypedMap(
+                                                        Object.class,
+                                                        eventData,
+                                                        EventDataKeys.TARGET_DATA_PAYLOAD,
+                                                        null),
+                                                request);
+                                final String content =
+                                        DataReader.optString(
+                                                eventData,
+                                                EventDataKeys.TARGET_CONTENT,
+                                                request.getDefaultContent());
+                                contentWithDataCallback.call(content, mboxPayloadMap);
+                            } else if (callback != null) {
+                                callback.call(
+                                        DataReader.optString(
+                                                eventData,
+                                                EventDataKeys.TARGET_CONTENT,
+                                                request.getDefaultContent()));
+                            }
+                        } catch (NullPointerException e) {
```

Simply catch all exceptions since the user code could throw anything.